test/rest.test.js

```diff
@@ -994,6 +994,37 @@ describe('strong-remoting-rest', function() {
         });
     });
 
+    it('should not flatten arrays for non-standard target type', function(done) {
+      var method = givenSharedStaticMethod(
+        function(arg, cb) { cb(null, { value: arg }); },
+        {
+          // Note: the string "array" is not a valid type
```

Comment:
See http://docs.strongloop.com/display/LB/LoopBack+types#LoopBacktypes-Arraytypes, these are the valid types:

 - `[String]`
 - `["String"]`
 - `[{type: String, length: 64}]`

In all cases, it's an array with a single item that describes the type of array items.